pkg/connector/mapping.go

```diff
@@ -0,0 +1,348 @@
+	}
+}
+
+func (tc *TwitterClient) GetUserInfoBridge(userId string) *bridgev2.UserInfo {
+	var userinfo *bridgev2.UserInfo
+	if userCacheEntry, ok := tc.userCache[userId]; ok {
+		userinfo = &bridgev2.UserInfo{
+			Name:        ptr.Ptr(tc.connector.Config.FormatDisplayname(userCacheEntry.ScreenName, userCacheEntry.Name)),
+			Avatar:      MakeAvatar(userCacheEntry.ProfileImageURL),
+			Identifiers: []string{fmt.Sprintf("twitter:%s", userCacheEntry.ScreenName)},
+		}
+	}
+	return userinfo
+}
+
+func (tc *TwitterClient) TwitterAttachmentToMatrix(ctx context.Context, portal *bridgev2.Portal, intent bridgev2.MatrixAPI, attachment *types.Attachment) (*bridgev2.ConvertedMessagePart, []int, error) {
+	var attachmentInfo *types.AttachmentInfo
+	var attachmentURL string
+	var mimeType string
+	var indices []int
+	var msgType bridgeEvt.MessageType
+	extraInfo := map[string]any{}
+	if attachment.Photo != nil {
+		// image attachment
+		attachmentInfo = attachment.Photo
+		mimeType = "image/jpeg" // attachment doesn't include this specifically
+		msgType = bridgeEvt.MsgImage
+		attachmentURL = attachmentInfo.MediaURLHTTPS
+		indices = attachmentInfo.Indices
+	} else if attachment.Video != nil || attachment.AnimatedGif != nil {
+		// video attachment
+		if attachment.AnimatedGif != nil {
+			attachmentInfo = attachment.AnimatedGif
+			extraInfo["fi.mau.gif"] = true
+			extraInfo["fi.mau.loop"] = true
+			extraInfo["fi.mau.autoplay"] = true
+			extraInfo["fi.mau.hide_controls"] = true
+			extraInfo["fi.mau.no_audio"] = true
+		} else {
+			attachmentInfo = attachment.Video
+		}
+		mimeType = "video/mp4"
+		msgType = bridgeEvt.MsgVideo
+
+		highestBitRateVariant, err := attachmentInfo.VideoInfo.GetHighestBitrateVariant()
+		if err != nil {
+			return nil, nil, err
+		}
+		attachmentURL = highestBitRateVariant.URL
+		indices = attachmentInfo.Indices
+	}
+
+	if attachmentInfo == nil || attachmentInfo.IDStr == "" {
+		return nil, nil, fmt.Errorf("missing attachment info")
+	}
+
+	attachmentBytes, err := DownloadPlainFile(ctx, tc.client.GetCookieString(), attachmentURL, "twitter attachment")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	content := convertTwitterAttachmentMetadata(attachmentInfo, mimeType, msgType, attachmentBytes)
+	err = uploadMedia(ctx, portal, intent, attachmentBytes, &content)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &bridgev2.ConvertedMessagePart{
+		ID:      networkid.PartID(fmt.Sprintf("attachment-%s", attachmentInfo.IDStr)),
+		Type:    bridgeEvt.EventMessage,
+		Content: &content,
+		Extra: map[string]any{
+			"info": extraInfo,
+		},
+	}, indices, nil
+}
+
+func uploadMedia(ctx context.Context, portal *bridgev2.Portal, intent bridgev2.MatrixAPI, data []byte, content *bridgeEvt.MessageEventContent) error {
+	mxc, file, err := intent.UploadMedia(ctx, portal.MXID, data, "", content.Info.MimeType)
+	if err != nil {
+		return err
+	}
+	if file != nil {
+		content.File = file
+	} else {
+		content.URL = mxc
+	}
```

Comment:
This is redundant. You can just do `content.URL, content.File, err = intent.UploadMedia(...)`